### ISSUE TITLE
cobbler system rename --newname test123 --name J31F695

### ISSUE THREAD

**Issue by hagbard-01** [cobbler/cobbler#3330]
### Describe the bug
I have build a live CD which uses the cobbler xmlrpc API to create a system record within cobbler, to have a unique name I use the serial number of the system. In my example below the serial is " J31F695", so the --name variable will have that as the system name.
If I rename the system, cobbler causes an exception. The json file in collections/systems doesn't exist anymore, but the original name is still in collections/profiles. The issue is not present if I create a system with a name let's say test and rename to test123, then everything works just fine. 

### Steps to reproduce

1. cobbler system  add --name J31F695 --profile almalinux8-x86_64
2. cobbler system  rename --name J31F695 --newname test123

result: 
exception on server: 'J31F695'

1. cobbler system  add --name test --profile almalinux8-x86_64
2. cobbler system  rename --name test --newname test123
3. cobbler system list 
    test123

### Expected behavior

new name in cobbler instead of an exception.
```
### Cobbler version
 cobbler --version
Cobbler 3.3.2
  source: ?, ?
  build time: Wed May  4 01:38:10 2022

Name         : cobbler
Version      : 3.3.2
Release      : 2.module_el8+14420+e999864d
Architecture : noarch
Size         : 2.7 M
Source       : cobbler-3.3.2-2.module_el8+14420+e999864d.src.rpm
Repository   : @System
From repo    : epel-modular
Summary      : Boot server configurator
URL          : https://cobbler.github.io/
License      : GPLv2+
Description  : Cobbler is a network install server. Cobbler supports PXE, ISO
             : virtualized installs, and re-installing existing Linux machines. The
             : last two modes use a helper tool, 'koan', that integrates with cobbler.
             : Cobbler's advanced features include importing distributions from DVDs
             : and rsync mirrors, kickstart templating, integrated yum mirroring, and
             : built-in DHCP/DNS Management. Cobbler has a XML-RPC API for integration
             : with other applications.

```
### Operating system
almalinux8

### Cobbler log
```
[Thread-50] 2023-01-18T09:25:00 - INFO | Exception occurred: <class 'AttributeError'>
[Thread-50] 2023-01-18T09:25:00 - INFO | Exception value: 'NoneType' object has no attribute 'name'
[Thread-50] 2023-01-18T09:25:00 - INFO | Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 3640, in _dispatch
    return method_handle(*params)

  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 2043, in xapi_object_edit
    self.remove_item(object_type, object_name, token, recursive=recursive)

  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 1262, in remove_item
    self.api.remove_item(what, name, delete=True, with_triggers=True, recursive=recursive)

  File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 462, in remove_item
    self.get_items(what).remove(ref.name, recursive=recursive, with_delete=delete, with_triggers=with_triggers)

  File "/usr/lib/python3.6/site-packages/cobbler/cobbler_collections/systems.py", line 77, in remove
    self.api.serialize()

  File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 1647, in serialize
    self._collection_mgr.serialize()

  File "/usr/lib/python3.6/site-packages/cobbler/cobbler_collections/manager.py", line 145, in serialize
    serializer.serialize(self._systems)

  File "/usr/lib/python3.6/site-packages/cobbler/serializer.py", line 83, in serialize
    storage_module.serialize(collection)

  File "/usr/lib/python3.6/site-packages/cobbler/modules/serializers/file.py", line 119, in serialize
    serialize_item(collection, x)

  File "/usr/lib/python3.6/site-packages/cobbler/modules/serializers/file.py", line 73, in serialize_item
    if not item.name:
```
### Additional information

SELinux is disabled, so it can't be a label issue.

**Comment by SchoolGuy** [comment]
Thanks for this bug report. I don't know how much effort it is for you but is it possible in your script that instead of `xapi_object_edit` you use the `rename_system` XML-RPC method? Your bug report is valid but if you are lucky the bug is only present in that particular function I want to get rid of in the short/mid term anyway...

**Comment by hagbard-01** [comment]
Hi @SchoolGuy , thanks for the quick reply. Sorry if I wasn't clear, the edit is broken in the cli already, that's why I posted and confirmed the behavior via the cobbler cli.  It appears to me that it has an issue with the name, if the name is something like test123, rename works. If the name is however something like J31F695, then the rename throws that exception.  You should be able to trigger the issue just by creating a system and try to rename by just using the cobbler cli.

**Comment by SchoolGuy** [comment]
@hagbard-01 The CLI is using the mentioned method and not `rename_system`. Thus my suggestion to use an alternative method.

See here:

https://github.com/cobbler/cobbler/blob/e9d837f80131f91463c837998b8d28064c1e6cd2/cobbler/cli.py#L2288-L2315

**Comment by hagbard-01** [comment]
Yes, and I triggered the issue when I used the cli. The system json file disappeared as well when rename is being executed via the cli, the system name is only still in the profile file present. As I evaluating cobbler right now, I have only 2 systems in there and will have a look as soon as I have the time to see where and why it fails.

**Comment by hagbard-01** [comment]
All right, it is a lower() in one of the classes, the issue isn't present if the name is set all lower case, then find_match in collections.py works as expected. I didn't find where it sets that yet, but I have a workaround for now. Since you rewrite the functionality  anyway I'm not sure if I should spend the time finding it and patching it eventually. I think if anyone else runs into that issue, they can just use all lower case names as well. Let me know, I'm more that happy to dig deeper into it, as time allows, but from my perspective it won't make much sense.

**Comment by SchoolGuy** [comment]
Yeah, I fully agree. Thanks for looking into this so detailed from your side. I will try to advance on my new server-decoupled CLI.
